crabs/bboxes_labelling/annotations_utils.py

```diff
@@ -79,6 +84,17 @@ def combine_multiple_via_jsons(
     dict_of_via_img_metadata = {}
     list_of_via_img_id_list = []
 
+    # Apply exclude pattern if required
+    if exclude_pattern:
+        list_json_files = [
+            js
+            for js in list_input_json_files
+            if not re.search(exclude_pattern, str(js))
+        ]
+    else:
+        list_json_files = list_input_json_files.copy()
```

Comment:
Any reason to copy the list? I'd probably just filter it in place...
```python
if exclude_pattern:
    not_excluded = lambda js: not re.search(exclude_pattern, str(js))
    list_input_json_files = [js for js in list_input_json_files if not_excluded(js)]
```

With a `lambda` because I'm a hipster.

Reply:
wooo fancy! ✨ me gusta

Reply:
but [Ruff disapproves](https://docs.astral.sh/ruff/rules/lambda-assignment/) 😢

Reply:
removed the copy but kept list comprehension

Reply:
Sigh. I love named lambdas, they're great, especially for readability.